main/src/library/String.flix

```diff
@@ -99,6 +100,1199 @@ namespace String {
+                        Some(s1, maxlen + 1)
+                    }
+                    case Some(ix2) => {
+                        let s1 = slice(s, ix, ix2);
+                        Some(s1, ix2 + sublen)
+                    }
+                }
+            };
+        List.unfold(step, 0)
+
+
+    ///
+    /// This is `indexOfLeft` with a start offset.
+    ///
+    /// Returns `None` if `sub` is the empty string.
+    ///
+    pub def indexOfLeftWithOffset(sub: String, offset: Int32, s: String): Option[Int32] =
+        import java.lang.String.indexOf(String, Int32);
+        if (isEmpty(sub))
+            None
+        else {
+            let ix = s.indexOf(sub, offset) as & Pure;
+            if (ix < 0) None else Some(ix)
+        }
+
+    ///
+    /// This is `indexOfRight` with a start offset.
+    ///
+    /// Returns `None` if `sub` is the empty string.
+    ///
+    pub def indexOfRightWithOffset(sub: String, offset: Int32, s: String): Option[Int32] =
+        import java.lang.String.lastIndexOf(String, Int32);
+        if (isEmpty(sub))
+            None
+        else {
+            let ix = s.lastIndexOf(sub, offset) as & Pure;
+            if (ix < 0) None else Some(ix)
+        }
+
+    ///
+    /// Split the string `s` into chunks of length `k`, the last chunk may be smaller than `k`.
+    ///
+    /// `k` should be greater than 0.
+    ///
+    pub def toChunks(k: Int32, s: String): List[String] =
+        if (k <= 0)
+            Nil
+        else
+            toChunksHelper(k, s)
+
+    ///
+    /// Helper function for `toChunks`.
+    ///
+    /// Precondition: size > 0
+    ///
+    def toChunksHelper(size: Int32, s: String): List[String] =
+        let len = length(s);
+        let step = pos -> match pos {
+            case i if i >= len          => None
+            case i if i + size >= len   => Some(sliceRight(s, i), len)
+            case i                      => Some(slice(s, i, i + size), i + size)
+        };
+        List.unfold(step, 0)
+
+    ///
+    /// Find the first instance of `sub` in string `s`, return a pair of the
+    /// prefix of string `s` up to `sub` and the rest of string `s` including `sub`.
+    ///
+    pub def breakOnLeft(sub: String, s: String): (String, String) =
+        match indexOfLeft(sub, s) {
+            case None => (s, "")
+            case Some(i) => (sliceLeft(s, i), sliceRight(s, i))
+        }
+
+    ///
+    /// Find the last instance of `sub` in string `s`, return a pair of the
+    /// initial string including `sub` and suffix from `sub`.
+    ///
+    pub def breakOnRight(sub: String, s: String): (String, String) =
+        let sublen = length(sub);
+        match indexOfRight(sub, s) {
+            case None => (s, "")
+            case Some(i) => (sliceLeft(s, i + sublen), sliceRight(s, i + sublen))
+        }
+
+    ///
+    /// Count the occurences of `sub` in string `s`.
+    ///
+    pub def count(sub: String, s: String): Int32 =
```

Comment:
"Usually" `count` would be a higher-order function "counting" the number of characters in the string that satisfy a predicate (i.e. like in list).

Reply:
Renamed to `countSubstring`.